### ISSUE TITLE
Help requested with optional member access in test code.

### ISSUE THREAD

**Issue by ChristopherZhong** [microsoft/pyright#2040]
**Describe the bug**
In a project, pyright was pinned to 1.1.123.
I'm trying to update the version to 1.1.153.
However, some of the unit tests are failing because of the optional member access error.

The following is a example to illustrate the problem.

sample.py

```python
from typing import Optional

class Sample:
    variable: str

def sample() -> Optional[Sample]:
    return Sample()
```

tests.py

```python
from unittest import TestCase
from sample import sample

class SampleTests(TestCase):
    def test_1(self) -> None:
        result = sample()

        self.assertIsNotNone(result)
        self.assertEqual('test', result.variable)
```

Running `npx pyright` produces the following result.

```
Searching for source files
Found 2 source files
/.../tests.py
  /.../tests.py:9:41 - error: "variable" is not a known member of "None" (reportOptionalMemberAccess)
1 error, 0 warnings, 0 infos 
Completed in 0.598sec
```

I understand why the error occurs but I think it should not be reporting the `reportOptionalMemberAccess` error for that line.
I can remove the error by adding a `assert result is not None`.
However, I think that adding an `assert result is not None` line to the tests is not a good workaround.
Is there a better way?

**Comment by JelleZijlstra** [comment]
You're seeing this because `reportOptionalMemberAccess` was turned on by default recently (#1989). Thus, a workaround for you could be to turn it off again in your configuration.

There is also #2007 asking for pyright to use `assertIsNotNone` as a None check, but the feature request was rejected for now.

**Comment by erictraut** [comment]
As discussed above, you have several options:

1. Use `assert result is not None` rather than `self.assertIsNotNone(result)`. The former is built into the language, so pyright understands its semantics, whereas the latter is a function call, and from a type perspective, it looks like any other function call to pyright, so it doesn't know that it should apply type narrowing.
2. Disable `reportOptionalMemberAccess` either globally for your project.
3. Disable `reportOptionalMemberAccess` locally to test files where you rely on this using a `# pyright: reportOptionalMemberAccess=false` comment at the top.

My my team's code base, we use option 1.
